lib/tests/csrfProtected.test.js

```diff
@@ -0,0 +1,152 @@
+import mockedAxios from "axios";
+import { createMocks } from "node-mocks-http";
+import { submitToAPI } from "../integration/helpers";
+import { logMessage } from "@lib/logger";
+import TestForm from "./testData";
+import { getCsrfToken } from "next-auth/client";
+import { csrfProtected } from "@lib/middleware/csrfProtected";
+
+jest.mock("axios");
+jest.mock("next-auth/client");
+
+const mockResponses = {
+  1: "test response", // TextField
+  2: "test response 2", // TextArea
+  4: "Manitoba", // DropDown
+  5: "Student", // Radio
+  6: [], // CheckBox
+  7: [
+    {
+      0: "Doe",
+      1: "John",
+      2: "CDS",
+      3: ["No"],
+    },
+    {
+      0: "Seat",
+      1: "Love",
+      2: "CHMC",
+      3: ["Yes", "Other"],
+    },
+  ],
+  8: "",
+};
+
+describe("Csrf Protection middleware", () => {
```

Comment:
Just a note that CSRF is usually only applied to state changing API calls (POST, PUT, DELETE)